src/components/register-in-contest.jsx

```diff
@@ -79,79 +105,64 @@ export default React.createClass({
     </div>;
   },
 
+  renderFormOrMessage(renderForm, formId) {
+    if (formId < this.props.user.registration_step_number) {
+      return this.renderSuccess();
+    } else if (formId === this.props.user.registration_step_number) {
+      return renderForm();
+    } else {
+      return this.renderUnavailableStep();
+    }
+  },
+
   renderSuccess() {
```

Comment:
as redenumi in `renderSuccessStep` sau `renderFinishedStep`. `renderSuccess` nu prea face referinta la step...`renderUnavailableStep` e super clar